### DISCUSSION TITLE
What is app_store_connect_api_key's 'in_house' option for?

### DISCUSSION THREAD

**Discussion by PhilippOtto89** [fastlane/fastlane#18084] [Q&A]
Hey everyone,

first of all thanks for reading my question in advance and of course thanks to all the contributors of fastlane to supply such an awesome tool. :)

Our team is switching from user/password auth to using an App Store Connect API Token in order to circumvent 2FA.
We have a developer subscription for both Enterprise and Non-Enterprise.

Fastlane's `app_store_connect_api_key` has an option `in_house` for which it says:

> Note: The App Store Connect API does not yet have the ability to determine if the team is App Store or Enterprise. The app_store_connect_api_key action and the fastlane API Key JSON file format allow for an optional in_house key as a temporary workaround.

I read that API Keys are not a thing for enterprise accounts (https://developer.apple.com/forums/thread/117282) and I can not find an option to create a key for enterprise accounts in Users & Access while for the Non-Enterprise Account I have the option.

When I run `match` with the enterprise team id and the non-enterprise token with `in_house: true` I see that `match` tries to access the profiles/certs of the non-enterprise account and not as intended of the enterprise account.

So my questions are:
1) What is `in_house: true` even used for today if there are no enterprise API tokens?
2) Is it possible to run `match` against enterprise certs/profiles with a non-enterprise API Token?

This question maybe goes into a similar direction:
https://github.com/fastlane/fastlane/issues/18033

Thanks again!

Best,
Philipp

**Comment by dinsen** [comment]
Hi @PhilippOtto89 
I just think joshdholtz have prepared the code for use with enterprise auth keys, so when Apple have added support for creating the keys on enterprise accounts, it will hopefully work right away.

**Reply by PhilippOtto89** [reply]
Hey @Dinsen, thanks for your answer. Ok so is this not supposed to work at the moment and only intended for future use? It was confusing me very much that this Bool is present and the documentation didn't contain any hint for it being not usable at the moment.

**Reply by dinsen** [reply]
I think so, since it's not possible to create an enterprise auth key. 

Not sure if someone from the Core team can give a better answer about this, but I will not spend my time on it until Apple have added support for creating enterprise auth keys.

**Reply by antoine20001** [reply]
Hello, I asked myself the same question and came to the same conclusion. 
Many of us are waiting for this feature for entreprise accounts.
